tests/test_georaster.py

```diff
@@ -750,28 +771,42 @@ def test_crop(self, data: list[str]) -> None:
             cropGeom[2] - rand_float * r.res[0],
             cropGeom[3] - rand_float * abs(r.res[1]),
         ]
-        r_cropped = r.crop(cropGeom2, inplace=False, mode="match_extent")
+        with warnings.catch_warnings():
```

Comment:
Could you add a comment for why UserWarnings are expected here?

Reply:
Also, I'm pretty sure that all our warnings are UserWarnings, so that means all of them get suppressed.

Reply:
Done, added a comment everywhere + the `message` argument with a regex pattern to only catch the desired warnings.